### PR TITLE
fix: bugsnag errors sprint-31

### PR DESCRIPTION
**Pull request by sanpj2292**
Signed-off-by: Sai Sankeerth <sanpj2292@github.com>

## Description of the change

[Google Adwords Remarketing Lists]
https://app.bugsnag.com/rudderlabs/rudder-transformer-error-reporter/errors/63a67990dcaa2500086ad048?event_id=63a6799000a2ec03dd930000&i=wh&m=nw&pivot_tab=event

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by saikumarrs — COMMENTED** [review]
I'd put each fix in individual PRs to keep the commit history and release notes clean.

File: src/v0/destinations/salesforce/utils.js [hunk 1]
```diff
@@ -27,7 +27,11 @@ const salesforceResponseHandler = (destResponse, sourceMessage, authKey) => {
   const { status, response } = destResponse;
 
   // if the response from destination is not a success case build an explicit error
-  if (!isHttpStatusSuccess(status) && status >= 400) {
+  if (
+    // This condition will pop-up when `invalid_grant` error is thrown from Salesforce
+    !Number.isInteger(status) ||
```
Comment (saikumarrs) [inline src/v0/destinations/salesforce/utils.js:32]: Are we sure this is the only scenario where the status is a non-number?

File: test/__tests__/data/salesforce_input.json [hunk 2]
```diff
@@ -793,7 +1168,45 @@
       "timestamp": "2020-01-27T17:50:57.109+05:30",
       "type": "identify",
       "userId": "1e7673da-9473-49c6-97f7-da848ecafa76"
-    }
+    },
+    "mockNetworkResponses": [
```
Comment (saikumarrs) [inline test/__tests__/data/salesforce_input.json:1172]: `Responses` is not the right terminology as both request and response data are covered here. `networkMockData` is a better name.

File: test/__tests__/data/salesforce_input.json [hunk 3]
```diff
@@ -793,7 +1168,45 @@
...
+      {
+        "type": "post",
```
Comment (saikumarrs) [inline test/__tests__/data/salesforce_input.json:1174]: I believe we can also include `type` into `reqParams`. So, the input becomes simple.

```
[
  {
    request: { ... },
    response: { ... }
  },
  {
    request: { ... },
    response: { ... }
  },
  ...
]
```

File: test/__tests__/salesforce.test.js [hunk 4]
```diff
@@ -38,33 +39,41 @@ const outputRouterMetadataFile = fs.readFileSync(
...
-      const routerOutput = await transformer.processRouterDest(inputRouterData);
-      expect(routerOutput).toEqual(expectedRouterData);
+  inputData.forEach(async (input, index) => {
+    it(`Payload - ${index}`, async () => {
+      try {
+        const output = await transformer.process(input);
+        expect(output).toEqual(expectedData[index]);
+      } catch (error) {
+        console.log(error);
+        expect(error.message).toEqual(expectedData[index].error);
```
Comment (saikumarrs) [inline test/__tests__/salesforce.test.js:55]: Let's use the `toThrow` Jest API instead.

**Review by saikumarrs — COMMENTED** [review]
LGTM overall.

**Comment by sanpj2292** [comment]
Not needed right now as we have done the same thing in #1776

**Pull request closed without merging**